### PR TITLE
Don't output passwords and tokens via AuthInfo toString

### PR DESCRIPTION
**Pull request by checketts**

### CODE REVIEW

File: client/src/main/scala/skuber/api/package.scala [hunk 1]
```diff
@@ -45,7 +45,26 @@ package object client {
...
      userName: Option[String] = None,
      password: Option[String] = None
-   )
+   ) {
+     override def toString : String = {
+       var result = new StringBuilder("AuthInfo(")
+       clientCertificate.getOrElse("") match {
+         case Left(cert) => result ++= "clientCertificate=" + cert + " "
+         case Right(cert) => result ++= "clientCertificate=<PEM masked> "
+         case _ =>
```
Comment (coryfklein) [inline client/src/main/scala/skuber/api/package.scala]: `Either` must be one of `Left` or `Right`, so the `case _ =>` is extraneous.

File: client/src/main/scala/skuber/api/package.scala [hunk 2]
```diff
@@ -45,7 +45,26 @@ package object client {
...
+       }
+       clientKey.getOrElse("") match {
+         case Left(cert) => result ++= "clientKey=" + cert + " "
+         case Right(cert) => result ++= "clientKey=<PEM masked> "
+         case _ =>
```
Comment (coryfklein) [inline client/src/main/scala/skuber/api/package.scala]: Ditto for here.

File: client/src/main/scala/skuber/api/package.scala [hunk 3]
```diff
@@ -45,7 +45,26 @@ package object client {
      token: Option[String] = None,
      userName: Option[String] = None,
      password: Option[String] = None
-   )
+   ) {
+     override def toString : String = {
+       var result = new StringBuilder("AuthInfo(")
+       clientCertificate.getOrElse("") match {
+         case Left(cert) => result ++= "clientCertificate=" + cert + " "
+         case Right(cert) => result ++= "clientCertificate=<PEM masked> "
```
Comment (coryfklein) [inline client/src/main/scala/skuber/api/package.scala]: Can we change this to read:

```
case Left(certPath: String) => result ++= "clientCertificate=" + certPath + " "
case Right(_: Array[Byte]) => result ++= "clientCertificate=<PEM masked> "
```

That makes it more clear what we are printing and what we are masking.

File: client/src/main/scala/skuber/api/package.scala [hunk 4]
```diff
@@ -45,7 +45,26 @@ package object client {
...
+         case _ =>
+       }
+       clientKey.getOrElse("") match {
+         case Left(cert) => result ++= "clientKey=" + cert + " "
+         case Right(cert) => result ++= "clientKey=<PEM masked> "
+         case _ =>
+       }
+       result ++= token.map("token="+_.replaceAll(".","*")+" ").getOrElse("")
```
Comment (coryfklein) [inline client/src/main/scala/skuber/api/package.scala:59]: Let's do `replaceAll(...).take(5)` to additionally mask length.

File: client/src/test/scala/skuber/model/AuthSpec.scala [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  "Auth toString doesn't mask username, certPath, keyPath" >> {
+    val auth = AuthInfo(
+      clientCertificate = Option(Left("certPath")),
+      clientKey = Option(Left("keyPath")),
+      userName = Option("aUser")
+    )
+    auth.toString mustEqual "AuthInfo(clientCertificate=certPath clientKey=keyPath userName=aUser )"
+  }
+
+}
```
Comment (coryfklein) [inline client/src/test/scala/skuber/model/AuthSpec.scala:37]: Is this file supposed to be part of the PR?

**Review by coryfklein — CHANGES_REQUESTED** [review]
Great idea. +1 if we clean up a few of the smaller things.

File: client/src/main/scala/skuber/api/package.scala (hunk 1 above)
Comment (checketts) [reply client/src/main/scala/skuber/api/package.scala]: The `_` was needed to match on the `getOrElse("")` of the `Option` (which is `String`)

File: client/src/test/scala/skuber/model/AuthSpec.scala (hunk 5 above)
Comment (checketts) [reply client/src/test/scala/skuber/model/AuthSpec.scala:37]: I thought I would add unit tests, so I included them in the PR.

Comment (coryfklein) [reply client/src/test/scala/skuber/model/AuthSpec.scala:37]: Ah, my bad. I mistook this for the `AuthInfo` definition due to the "David O'Riordan" authorship note. Shouldn't that be "Clint Checketts"?

**Comment by checketts** [comment]
Thanks for the feedback @coryfklein . I pushed those fixes

**Review by coryfklein — APPROVED** [review]

**Pull request merged**